packages/ui/src/mainbar.tsx

```diff
@@ -0,0 +1,82 @@
+'use client';
+
+import { useState } from 'react';
+import MenuIcon from '@mui/icons-material/Menu';
+import {
+	AppBar,
+	Drawer,
+	IconButton,
+	InputBase,
+	Toolbar,
+	Typography,
+} from '@mui/material';
+import Link from 'next/link';
+
+import Sidebar from './sidebar';
+import SigninWithGoogleButton from './signInWithGoogleButton';
+
+const Navbar = () => {
+	const [sidebarOpen, setSidebarOpen] = useState(false);
+
+	const handleDrawerToggle = () => {
+		setSidebarOpen(!sidebarOpen);
+	};
+
+	return (
+		<>
+			<AppBar position="static" component="nav" className="appbar">
+				<Toolbar>
+					<IconButton
+						color="inherit"
+						aria-label="open drawer"
+						edge="start"
+						onClick={handleDrawerToggle}
+					>
+						<MenuIcon />
+					</IconButton>
+					<Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
+						<Link
+							href="/"
+							style={{ textDecoration: 'none', color: 'white', marginLeft: 10 }}
+						>
+							MetaCMS
+						</Link>
+					</Typography>
+					<div style={{ position: 'relative' }}>
+						<InputBase
+							placeholder="Search..."
+							inputProps={{ 'aria-label': 'search' }}
+							sx={{
+								color: 'inherit',
+								'& .MuiInputBase-input': { width: '150px' },
+								mx: 3,
+							}}
+						/>
+					</div>
+					<SigninWithGoogleButton />
+				</Toolbar>
+			</AppBar>
+			<nav>
+				<Drawer
+					variant="temporary"
+					open={sidebarOpen}
+					onClose={handleDrawerToggle}
+					ModalProps={{
+						keepMounted: true,
+					}}
+					sx={{
+						display: 'block',
+						'@media (min-width:600px)': {
+							display: 'block',
+						},
+						'& .MuiDrawer-paper': { boxSizing: 'border-box', width: 240 },
+					}}
```

Comment:
don't put this object directly here, would be something like this:
```typescript
export const Navbar = () => {
  return (
    <Fragment>
      <Drawer ... sx={styles.drawerContainer} />
    </Fragment>
  );
};

const styles = {
  drawerContainer: {
    display: 'block',
      '@media (min-width:600px)': {
        display: 'block',
      },
     '& .MuiDrawer-paper': { boxSizing: 'border-box', width: 240 },
  },
}
```

always great to keep XML markup as lean as possible, this help us imagine the hierarchy of the elements-tree much better.